### DISCUSSION TITLE
Disable updating to snapshots

### DISCUSSION THREAD

**Discussion by marc-ernst** [renovatebot/renovate#14961] [Request Help] [answered]
Hi,
is this a valid "packageRules" syntax to disable snapshot updates:

```json
"packageRules": [
    {
       "description": "Minor lib updates",
       "groupName": "libs update",
       "managers": ["regex"],
       "matchUpdateTypes": [ "minor", "patch" ],
       "excludePackagePrefixes": ["adoptium"]
    },
    {
       "description": "Minor JDK updates",
       "groupName": "JDK update",
       "managers": ["regex"],
       "matchUpdateTypes": [ "minor", "patch" ],
       "matchPackagePrefixes": [ "adoptium" ]
    },
    {
      "description": "No updates for SNAPSHOTS",
      "allowedVersions": "/\\-SNAPSHOT$/",
      "enabled": false
    }
]
```

I want to prevent snapshot updates on all groups but mainly on the first one (libs).

Thanks,
Marc

**Comment by rarkins** [comment, marked as answer]
No, it's not valid. Use matchDatasources=maven and allowedVersions. Don't use enabled

**Reply by marc-ernst** [reply]
Thanks for the fast reply. Should I use matchDatasources although I use a regexMatcher and have both datasource=github-releases / datasource=maven ? I tried with allowedVersions but could not find out how to prevent major updates ("allowedVersions": "!/\\-SNAPSHOT$/") because matchUpdateTypes cannot be combined with allowedVersions.

**Reply by marc-ernst** [reply]
Can I use preset :disableMajorUpdates together with allowedVersions?

**Reply by rarkins** [reply]
You can use whatever you need to in order to match the packages. Datasources, managers, names or patterns, etc.

**Reply by marc-ernst** [reply]
For some reasons the updates are not grouped anymore. I've got two patches for two libraries:

"packageRules": [
    {
       "description": "Minor ebr library updates",
       "groupName": "ebr library updates",
       "managers": [ "regex" ],
       "matchUpdateTypes": [ "minor", "patch" ],
       "matchDatasources": [ "maven" ],
       "excludePackagePrefixes": [ "adoptium" ]
    },
    {
       "description": "Minor JDK updates",
       "groupName": "JDK update",
       "managers": [ "regex" ],
       "matchUpdateTypes": [ "minor", "patch" ],
       "matchDatasources": [ "adoptium-java" ],
       "matchPackagePrefixes": [ "adoptium" ]
    }
  ],

**Reply by rarkins** [reply]
Please create a public reproduction we can look at

**Reply by viceice** [reply]
remove the `groupName` settings, those will cause branch name changes. 😉

**Reply by marc-ernst** [reply]
I removed the group name but this did not help and it still tries to update snapshots: https://github.com/marc-ernst/maven-libs/pulls

**Reply by rarkins** [reply]
I don't see any snapshot updates in the list of pulls. which are you referring to?

**Reply by marc-ernst** [reply]
Yes, sorry, the snapshot updates are not there anymore but the updates are not grouped. Is it because they have different data sources and I've to add them in addition to maven?

**Reply by rarkins** [reply]
Your config doesn't include any `groupName` now, so I wouldn't expect any grouping unless packages come from the same monorepo. Recommend you add new `packageRules` just for the grouping you want to do.

**Reply by marc-ernst** [reply]
ok, adding again the groupname and all data sources to one package rule solved the issue.

**Comment by ahoehma** [comment]
Please guys can u share a complete example here. I have the same challenge: I need only updates for release-versions in maven multimodule project.
